Bluetooth: Controller: Fix Data Length Request Cmd parameter checking

cvinayak opened the pull request:
Fix Data Length Request CMD parameter checking to not
restrict invalid maximum Tx time values. This addresses
conformance test cases that provide maximum Tx time values
of 2128 us when not supporting Coded PHY in the Controller
implementation.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Fix Data Length Request Cmd parameter checking

Fix Data Length Request CMD parameter checking to not
restrict invalid maximum Tx time values. This addresses
conformance test cases that provide maximum Tx time values
of 2128 us when not supporting Coded PHY in the Controller
implementation.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

erbr-ot commented on subsys/bluetooth/controller/ll_sw/ull_conn.c:
```diff
@@ -4605,6 +4610,15 @@ static inline void event_phy_upd_ind_prep(struct ll_conn *conn,
 			conn->llcp_ack = conn->llcp_req;
 		}
 
+		/* supported PHYs mask */
```
wondering why we should wait until instant to check this? Why not do it when we check the _IND?

cvinayak replied:
There are two allowed pass criteria:
1. To disconnect if peer uses unsupported PHY
2. Continue the connection on old PHY

For backwards compatibility and not to change the controller behavior, option 2 is selected. 
Under option 2, the new PHY is committed at the instant, and this is where the decision to not have a PHY update event towards Host is implemented, hence I added the checks here.

erbr-ot replied:
OK, I get that, and I also now understand that in case only one of the PHYs are suggesting an unsupported PHY we need to process the other one, so we have to handle this at instant.

erbr-ot replied:
And this needs to be handled in the refactored LLCP as well

cvinayak commented:
@erbr-ot and @wopu-ot failing EDTT test needs looking into.

Thalley commented on subsys/bluetooth/controller/ll_sw/ull_conn.c:
```diff
@@ -677,6 +668,20 @@ uint32_t ll_length_req_send(uint16_t handle, uint16_t tx_octets,
 
+#if defined(CONFIG_BT_CTLR_PHY)
+#if defined(CONFIG_BT_CTLR_PHY_CODED)
+	uint16_t tx_time_max =
+			PDU_DC_MAX_US(LL_LENGTH_OCTETS_TX_MAX, PHY_CODED);
+#else /* !CONFIG_BT_CTLR_PHY_CODED */
+	uint16_t tx_time_max =
```
```suggestion
	const uint16_t tx_time_max =
			PDU_DC_MAX_US(LL_LENGTH_OCTETS_TX_MAX, PHY_CODED);
#else /* !CONFIG_BT_CTLR_PHY_CODED */
	const uint16_t tx_time_max =
```

Thalley commented on subsys/bluetooth/controller/ll_sw/pdu.h:
```diff
@@ -84,13 +84,9 @@
-#define PDU_DC_PAYLOAD_TIME_MAX 2120
-#endif
-
+/* Data channel maximum payload time */
```
```suggestion
/* Data channel maximum payload time in us*/
```

Thalley reviewed (changes requested):
Minor comment otherwise LGTM

cvinayak committed:
tests: Bluetooth: EDTT: Fix incorrectly disabled legacy LLCP tests

Fix incorrectly disabled tests that had been passing on
legacy LLCP implementation in the Controller.

Regression introduced in commit f023b5f611df ("Bluetooth:
controller: push topic branch to main").

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented:
> Yeah, the EDTT test script is probably not happy about the fact that we now clamp the max_tx_time with no change resulting, even though the py-script expects a change. This test case has been commented out for LEGACY due to the error in the DLE handling (re. 1/2MPHYs in the calc of max times) that was in the legacy implementation. So I suspect that is why it 'only' fails for the refactored LLCP.

@erbr-ot infact the failing test remains disabled for new LLCP, I now have disabled it for legacy LLCP due to this PR.  I have also, re-enables tests that remain to pass but was incorrectly disabled due to new LLCP failure during the new LLCP development.

ppryga-nordic reviewed (commented):
LGTM
